### PR TITLE
Optimizer: When u flag is set, convert surrogate pairs to single unicode code point

### PR DESCRIPTION
**Pull request by Golmote**
`/\ud83d\ude80/u` -> `/\u{1f680}/u`

### CODE REVIEW

File: src/optimizer/transforms/char-surrogate-pair-to-single-unicode-transform.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module.exports = {
+  shouldRun(ast) {
+    return ast.flags.includes('u');
+  },
+  Char(path) {
+    const {node} = path;
+    if (node.kind !== 'unicode' || !node.isSurrogatePair || isNaN(node.codePoint)) {
+      return;
+    }
+    node.value = `\\u{${node.codePoint.toString(16).toLowerCase()}}`;
```
Comment (mathiasbynens) [inline src/optimizer/transforms/char-surrogate-pair-to-single-unicode-transform.js]: `toLowerCase()` is redundant here.

Comment (Golmote) [reply src/optimizer/transforms/char-surrogate-pair-to-single-unicode-transform.js]: Updated! Thanks.

**Comment by DmitrySoshnikov** [comment]
Hey guys, I’m currently on a small vacation, will take a look at all PRs once have a chance, and will be at laptop.

**Comment by Golmote** [comment]
@DmitrySoshnikov No problem, take your time. Enjoy your vacation!

**Comment by DmitrySoshnikov** [comment]
Nice one!

**Pull request merged**